rollout/controller.go

```diff
@@ -353,6 +353,12 @@ func (c *Controller) syncHandler(key string) error {
 	r := remarshalRollout(rollout)
 
 	if err := c.refResolver.Resolve(r); err != nil {
+		invalidSpecCond := conditions.NewRolloutCondition(v1alpha1.InvalidSpec, corev1.ConditionTrue, conditions.InvalidSpecReason, err.Error())
+		cr := r.DeepCopy()
+		cr.Status.Conditions = append(cr.Status.Conditions, *invalidSpecCond)
+		_, err = c.argoprojclientset.ArgoprojV1alpha1().Rollouts(r.Namespace).UpdateStatus(
+			context.TODO(), cr, metav1.UpdateOptions{},
+		)
```

Comment:
This seems like it would continue to add the InvalidSpec condition on every reconciliation. Can you make it so it doesn't continuously add to the condition list?

Also, a unit test would be nice.

Reply:
Thanks for reviewing! I will update the PR.

Reply:
@jessesuen , add some unit tests, but still failed to meet the 80% threshold